### DISCUSSION TITLE
Compressing database / support for vacuum

### DISCUSSION THREAD

**Discussion by foondadom** [capacitor-community/sqlite#595] [Q&A]
I have an Android database from which I've removed a bunch of base64 strings. As expected, the db does not decrease in size automatically, but I would like to vacuum the database once the base64 has been removed.

I've tried running the vacuum command (db.exec) on the open database which tells me that vacuum cannot be run while there's a transaction. I've also tried running the command before the db is open which naturally results in a 'no database open' error.

I've also tried running executeSet directly on the SQLitePlugin with the vacuum command, but again the database needs to be open and connected.

`const vacuumSet = [ { statement: "VACUUM;", values: [] } ];
await sqlitePlugin.executeSet(vacuumSet);`

Questions:
1. Is running an explicit vacuum command the best way to go?
2. I've seen mentions that auto_vacuum can also be set, how would I do this?
3. In the jeepsqlite documentation, it seems as though vacuum can be added to a transaction:
`let delUsers = "DELETE FROM users;";
              delUsers += "VACUUM;";
              ret = await jeepSqlite.execute({database: "testNew", statements: delUsers, transaction: false});`

Is there a better way to achieve database optimisation?

**Comment by jepiqueau** [comment]
@foondadom 
      // initialize the connection
      db = await this._sqlite.createConnection(YOUR_DBNAME, false, 'no-encryption', 1)
      // open your database
      await db.open();
     // execute the VACUUM without transaction
     ret = await db.execute('VACUUM',false);
     // close the connection
     await this._sqlite.closeConnection(YOUR_DBNAME);
